File: LeetCode/Easy/0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode middleNode(ListNode head) {
        int count = 1;
        ListNode temp = head;
        while (temp.next != null) {
            temp = temp.next;
            count++;
        }

        int mid = count / 2 + 1;

        for (int i = 1; i < mid; i++) {
            head = head.next;
        }
        
        return head;
    }
}

File: LeetCode/Easy/0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var middleNode = function(head) {
    let temp = head;
    let count = 0;
    while(temp.next) {
        temp = temp.next;
        count++;
    }
    
    count /= 2;
    
    while(count-- > 0) {
        head = head.next;
    }
    
    return head
};